### PR TITLE
feat: use shutdown height as sync stop height

### PR DESCRIPTION
**Pull request by Longarithm**
If one wants to take old backup to run some blocks on top of its head, node will first run header sync, and its default behaviour is to download **all** headers from chain. If backup is couple days old, this takes a while.

The solution is to use `expected_shutdown` config, which will stop header sync after shutdown height is reached and let the node download and process blocks. I tested this on my node, it worked nicely, allowing to change shutdown height on the fly.

### CODE REVIEW

File: chain/client/src/sync/header.rs [hunk 1]
```diff
@@ -147,8 +154,10 @@ impl HeaderSync {
         self.syncing_peer = None;
         // Pick a new random peer to request the next batch of headers.
         if let Some(peer) = highest_height_peers.choose(&mut thread_rng()).cloned() {
-            // TODO: This condition should always be true, otherwise we can already complete header sync.
-            if peer.highest_block_height > header_head.height {
+            let shutdown_height = self.shutdown_height.get();
```
Comment (wacban) [inline chain/client/src/sync/header.rs:157]: nit: I would put the unwrap_or here.

**Review by wacban — COMMENTED** [review]
I like the idea but `expected_shutdown` is taken already and has a different meaning. Can you make this a separate mutable config?

**Comment by Longarithm** [comment]
> I like the idea but expected_shutdown is taken already and has a different meaning. Can you make this a separate mutable config?

I don't introduce new parameter, I actually reuse existing expected_shutdown from config.json.
I think it is perfect usecase, because we don't want to sync after it anyway.

**Review by wacban — APPROVED** [review]
LGTM

**Comment by wacban** [comment]
> > I like the idea but expected_shutdown is taken already and has a different meaning. Can you make this a separate mutable config?
> 
> I don't introduce new parameter, I actually reuse existing expected_shutdown from config.json. I think it is perfect usecase, because we don't want to sync after it anyway.

Gotcha, my idea was to have one config for one meaning. Then you have more control over it as you can set each independently. Anyone since it's only for debugging I'm totally fine with this as is.

**Pull request merged**